mir_perception/mir_perception_utils/common/src/pointcloud_utils.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020 Bonn-Rhein-Sieg University
+ *
+ * Author: Mohammad Wasil
+ *
+ */
+#include <mir_perception_utils/pointcloud_utils.h>
+#include <pcl/common/centroid.h>
+#include <pcl/common/io.h>
+#include <pcl/PointIndices.h>
+#include <pcl/filters/extract_indices.h>
+
+using namespace mir_perception_utils;
+
+unsigned int pointcloud::centerPointCloud(const PointCloud &cloud_in,
+                             PointCloud &centered_cloud)
+{
+    if (cloud_in.empty ())
+        return (0);
+
+    pcl::copyPointCloud(cloud_in, centered_cloud);
+        
+    Eigen::Vector4f centroid;
+    pcl::compute3DCentroid(centered_cloud, centroid);
+ 
+    unsigned int point_count;
+    // Check if the data is dense, means there may be NaN or Inf values
+    if (centered_cloud.is_dense)
+    {
+        point_count = static_cast<unsigned int> (centered_cloud.points.size ());
+        // For each point in the cloud
+        for (auto& point: centered_cloud)
+        { 
+            point.x = point.x - centroid[0];
+            point.y = point.y - centroid[1];
+            point.z = point.z - centroid[2];
+        }
+    }
+    // Check NaN or Inf values, that could exist
+    else
+    {
+        point_count = 0;
+        for (auto& point: centered_cloud)
+        {
+            // Check if the point is invalid
+            if (!isFinite (point))
+                continue;
+
+            point.x = point.x - centroid[0];
+            point.y = point.y - centroid[1];
+            point.z = point.z - centroid[2];
+            ++point_count;
+        }
+    }
+    return (point_count);
+}
+
+unsigned int pointcloud::padPointCloud(PointCloud::Ptr &cloud_in,
+                                       int num_points)
+{
+    if (cloud_in->empty ())
+        return (0);
+    
+    unsigned int point_count;
+    point_count = static_cast<unsigned int> (cloud_in->size ());
+    
+    //ToDo: check if the cloud size > 2048, if so, downsample
```

Comment:
isn't this what you're doing here?

Reply:
yes,, forgot to remove todo. haha
I will update it and merge it